iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -72,29 +115,55 @@ static LogicalResult setContractConfig(FuncOp entryPoint, linalg::LinalgOp op) {
   int64_t tileX = 2;
   int64_t tileY = 256;
   int64_t tileK = 4;
+  IREE::Codegen::DispatchLoweringPassPipeline pipeline =
+      IREE::Codegen::DispatchLoweringPassPipeline::LLVMGPUMatmulSimt;
   SmallVector<int64_t, 3> workgroupSize = {2 * cudaWarpSize, 1, 1};
   bool isStaticSize =
       sizeM != ShapedType::kDynamicSize && sizeN != ShapedType::kDynamicSize;
   if (isStaticSize) {
-    // Special case for very small matrices.
-    if (sizeM * sizeN <= cudaWarpSize) {
-      tileX = sizeN;
-      tileY = sizeM;
-      workgroupSize = {sizeM, sizeN, 1};
+    bool foundTensorCoreConfig = false;
+    if (supportsTensorCore(entryPoint, op)) {
+      SmallVector<TileWorkgroupSizePair> TCtileSizeConfig;
+      getTensorCoreConfig(TCtileSizeConfig);
+      // Pick the best configuration where the original shape is aligned on the
+      // tile size.
+      for (TileWorkgroupSizePair &config : TCtileSizeConfig) {
```

Comment:
Nit : This is why I dislike side data structures that are just carrying information to just populate another data structure. It adds cognitive overhead to reading the code.

Reply:
The goal is to separate the configuration data from the logic. I don't really have a better solution at the time. Do you have any suggestions?

Reply:
I left comment above.